webplatformcompat/renderers.py

```diff
@@ -1,244 +1,242 @@
+
+        converted = self.dict_class(((main_resource, attributes),))
+        if link_ids:
+            converted[main_resource]['links'] = link_ids
+        if link_patterns:
+            converted['links'] = link_patterns
+        if linked:
+            converted['linked'] = linked
+        if meta:
+            converted['meta'] = meta
+        return converted
+
+    def convert_paginated(self, data, request):
+        """Convert paginated data to JSON API RC1.
+
+        The pagination data is moved to the "meta" key, and the
+        paginated "results" are split into resources and link data.
+        """
+        keys = set(data.keys())
+        is_paginated = keys >= set(('count', 'next', 'previous', 'results'))
+        converted = self.dict_class()
+
+        assert is_paginated
+        converted, resource = self.convert_list(data['results'], request)
+        pagination = self.dict_class((
+            ('previous', data['previous']),
+            ('next', data['next']),
+            ('count', data['count']),
+        ))
+        converted['meta'] = self.dict_class((
+            ('pagination', self.dict_class((
+                (resource, pagination),
+            ))),
+        ))
+        return converted
+
+    def convert_list(self, return_list, request):
+        """Convert a ReturnList to JSON API RC1 format.
+
+        The list data is moved under the resource name, and the link metadata
+        is collected under the "links" key, using field data from the
+        ReturnList's attached ListSerializer, and using the request to build
+        full URLs.
+        """
+        assert hasattr(return_list, 'serializer'), "Must be ReturnList"
+        serializer = return_list.serializer.child
+        fields_extra = serializer.get_fields_extra()
+        main_resource = fields_extra['id']['resource']
+
+        converted_items = []
+        converted_links = None
+        for item in return_list:
+            converted_item = self.convert_standard(
+                item, fields_extra, main_resource, request)
+            converted_items.append(converted_item[main_resource])
+            if converted_links is None:
+                converted_links = converted_item.get('links')
+
+        converted = self.dict_class(((main_resource, converted_items),))
+        if converted_links:
+            converted['links'] = converted_links
+        return converted, main_resource
+
+    def convert_link(self, name, value, field_extra, prefix, request):
+        """Convert an ID or ID list to the JSON API RC1 format.
+
+        Return is a tuple:
+        * pattern - The pattern dictionary
+        * link_id - The converted ID (or ID list, or None)
+        """
+        link = field_extra['link']
+        if value is None:
+            link_id = None
+        elif link in ('from_many', 'to_many'):
+            link_id = [str(pk) for pk in value]
+        elif link == 'self':
+            link_id = str(value)
```

Comment:
I can't tell if it's fishy that this is the same as the `else:` clause or not. Is that intended? If so, do you need this clause?